js/background/background.js

```diff
@@ -736,6 +736,76 @@ class Steam {
 Steam._dynamicstore_promise = null;
 Steam._supportedCurrencies = null;
 
+// When redirecting to Steam Community, if the game does not actually exist, onHeadersReceived would normally trigger an infinite redirection loop. To avoid that, we store the IDs of the requests that have already been redirected here.
+let webRequests = [];
+
+function onHeadersReceived(details) {
+    let response = {};
+
+    if (details.statusCode === 302 && webRequests.indexOf(details.requestId) < 0) {
+        let matches = details.url.match(/(app|sub)\/(\d+)/);
+
+        if (matches) {
+            let type = matches[1];
+            let id = matches[2];
+
+            switch (SyncedStorage.get('redirect_removed_to')) {
+                case 'community':
+                    response.redirectUrl = `https://steamcommunity.com/${type}/${id}`;
+                    break;
+                case 'steamdb':
+                    response.redirectUrl = `https://steamdb.info/${type}/${id}`;
+                    break;
+                case 'custom':
+                    response.redirectUrl = SyncedStorage.get('redirect_removed_custom')
+                        .replace(/\[TYPE]/, type)
+                        .replace(/\[ID]/, id);
+                    break;
+                default:
+                    break;
+            }
+
+            webRequests.push(details.requestId);
+        }
+    }
+
+    return response;
+}
+
+function addWebRequestListener() {
+    let filter = {
+        types: ['main_frame'], // Only requests triggered by the user through regular navigation.
+        urls: ['*://*.store.steampowered.com/app/*', '*://*.store.steampowered.com/sub/*'],
+    };
+    let extraInfoSpec = ['blocking'];
+
+    if (Info.browser === 'chrome') {
+        extraInfoSpec.push('extraHeaders'); // Required since Chrome v72.
```

Comment:
Where did you find this?

Reply:
https://developer.chrome.com/extensions/webRequest There are a ton of warnings about things that changed on Chrome v72 and Chrome v79 below the event definitions.

Reply:
I've seen this, but we're not using any of the headers that this option provides, if I read it correctly

Reply:
I think this is what applies:

```
Starting from Chrome 72, if you need to modify responses before Cross Origin Read Blocking (CORB) can block the response, you need to specify 'extraHeaders' in opt_extraInfpSpec. 
```

Not sure. What I know for sure is that as soon as I added `extraHeaders`, it started redirecting normally on Chrome.